### PR TITLE
fix mout vulnerability by replacing credential package with credentials

### PR DESCRIPTION
**Pull request by ETLaurent**
## Summary

* Replace [`credential`](https://www.npmjs.com/package/credential) package with [`credentials`](https://www.npmjs.com/package/credentials) to fix the [`mout` Prototype Pollution vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7792).

## What are the specific steps to test this change?

- link apostrophe into a project (testbed for instance)
- test login with a correct password -> should log in
- test login with a wrong password -> should not log in

⚠️ former users that were created before this code change (i.e with the credential package) should still be able to log in with their password.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [x] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: test/users.js [hunk 1]
```diff
@@ -98,6 +98,27 @@ describe('Users', function() {
...
+    user.username = 'olduser';
+    user.password = 'passwordThatWentThroughOldCredentialPackageHashing';
+    user.email = 'old@user.com';
+    user.role = 'admin';
+
+    await apos.user.insert(req, user);
+
+    // A password hash that were generated by the former credential package:
+    const oldPasswordHashSimulated =
+      '{"hash":"/1GntJjtkMY1iPmQY1gn9f3bOZ5tb2qFL+x4qsDerZq2JL8+12TERR4/xqh246wBb+QJwwIRsF/6E+eccshsLxT/","salt":"GJHukLNaG6xDgdIpxVOpqV7xQLQM7e5xnhDW7oaUOe7mTicr7Ca76M4uUJalN/cQ68CE9O7yXZ5WJOz4RN/udcX0","keyLength":66,"hashMethod":"pbkdf2","iterations":2853053}';
```
Comment (ETLaurent) [inline test/users.js:115]: Editing userSafe with a passwordHash generated with the old `credential` package

File: test/users.js [hunk 2]
```diff
@@ -98,6 +98,27 @@ describe('Users', function() {
...
+
+    await apos.user.safe.update({ username: 'olduser' }, { $set: { passwordHash: oldPasswordHashSimulated } });
+    await apos.user.verifyPassword(user, 'passwordThatWentThroughOldCredentialPackageHashing');
+    await apos.user.safe.remove({ username: 'olduser' });
```
Comment (ETLaurent) [inline test/users.js:119]: remove userSafe for scoping

**Review by boutell — CHANGES_REQUESTED** [review]
💯 just make clear in the changelog that there was never an actual vulnerability

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Fixes
+
+* Replace [`credential`](https://www.npmjs.com/package/credential) package with [`credentials`](https://www.npmjs.com/package/credentials) to fix the [`mout` Prototype Pollution vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7792).
```
[the remark this answers is not in the document]
Comment (ETLaurent) [reply CHANGELOG.md]: Used your phrase as it is very clear

**Review by boutell — APPROVED** [review]

**Pull request merged**